modules/hostap/src/supp_api.c

```diff
@@ -194,6 +194,26 @@ static int wpa_supp_supported_channels(struct wpa_supplicant *wpa_s, uint8_t ban
 	return 0;
 }
 
+static int wpa_supp_band_chan_compat(struct wpa_supplicant *wpa_s, uint8_t band, uint8_t channel)
+{
+	struct hostapd_hw_modes *mode = NULL;
+	int i;
+
+	mode = get_mode_by_band(wpa_s, band);
+	if (!mode) {
+		wpa_printf(MSG_ERROR, "Unsupported or invalid band: %d", band);
+		return -EINVAL;
+	}
+
+	for (i = 0; i < mode->num_channels; i++) {
+		if (mode->channels[i].freq == channel) {
+			return mode->channels[i].freq;
+		}
+	}
+
```

Comment:
Is it better to have a error log here to indicate band-channel compatibility check failure.

Reply:
Not only log, we need to propagate these error to the Application.